eth/gasprice/gasprice_test.go

```diff
@@ -257,223 +245,135 @@ func testGenBlock(t *testing.T, tip int64, numTx int) func(int, *core.BlockGen)
+				Gas:       ethparams.TxGas,
+				GasFeeCap: feeCap,
+				GasTipCap: txTip,
+				Data:      []byte{},
 			})
 			tx, err := types.SignTx(tx, signer, key)
 			require.NoError(t, err, "failed to create tx")
 			b.AddTx(tx)
 		}
-	})
-	defer backend.teardown()
-
-	oracle, err := NewOracle(backend, config)
-	require.NoError(t, err)
-
-	_, err = oracle.SuggestPrice(context.Background())
-	require.NoError(t, err)
+	}
 }
 
-// NOTE: [Oracle.SuggestTipCap] does NOT simply return the "required" (minimum) tip.
-// The oracle computes a percentile of recent required tips (not observed on-chain tips)
-// within a time/blocks lookback window and applies a small floor (e.g., 1 wei in tests):
-//
-//	suggested = max(floor, recent-required-percentile)
-//
-// After Granite, BlockGasCost is 0 and per-block required tips are 0, so the oracle
-// suggestion equals the floor (1 wei) in steady state, regardless of high on-chain tips.
-// The cases below exercise behavior across forks using the same percentile logic and floor.
-func TestSuggestTipCapMaxBlocksLookback(t *testing.T) {
+func TestSuggestTipCap(t *testing.T) {
 	cases := []struct {
-		chainConfig *params.ChainConfig
+		name        string
+		numBlocks   int
+		genBlock    func(int, *core.BlockGen)
 		expectedTip *big.Int
 	}{
-		// TODO: remove Fortuna case when we activate Granite
 		{
-			chainConfig: params.TestFortunaChainConfig,
-			expectedTip: big.NewInt(3),
+			name:        "simple_latest_no_tip",
+			numBlocks:   3,
+			genBlock:    testGenBlock(t, 0, 80),
+			expectedTip: DefaultMinPrice,
+		},
+		{
+			name:        "simple_latest_1_gwei_tip",
+			numBlocks:   3,
+			genBlock:    testGenBlock(t, 1, 80),
+			expectedTip: big.NewInt(1 * params.GWei),
+		},
+		{
+			name:        "simple_latest_100_gwei_tip",
+			numBlocks:   3,
+			genBlock:    testGenBlock(t, 100, 80),
+			expectedTip: big.NewInt(100 * params.GWei),
+		},
+		{
+			name:        "simple_floor_latest_1_gwei_tip",
+			numBlocks:   3,
+			genBlock:    testGenBlock(t, 1, 80),
+			expectedTip: big.NewInt(1 * params.GWei),
+		},
+		{
+			name:        "simple_floor_latest_100_gwei_tip",
+			numBlocks:   3,
+			genBlock:    testGenBlock(t, 100, 80),
+			expectedTip: big.NewInt(100 * params.GWei),
+		},
+		{
+			name:        "max_tip_cap",
+			numBlocks:   200,
+			genBlock:    testGenBlock(t, 550, 80),
+			expectedTip: DefaultMaxPrice,
+		},
+		{
+			name:        "single_transaction_with_tip",
+			numBlocks:   3,
+			genBlock:    testGenBlockWithTips(t, []int64{100}),
+			expectedTip: big.NewInt(100 * params.GWei),
+		},
+		{
+			name:        "three_transactions_with_odd_count_tips",
+			numBlocks:   3,
+			genBlock:    testGenBlockWithTips(t, []int64{10, 20, 30}),
+			expectedTip: big.NewInt(20 * params.GWei),
+		},
+		{
+			name:        "four_transactions_with_even_count_tips",
+			numBlocks:   3,
+			genBlock:    testGenBlockWithTips(t, []int64{10, 20, 30, 40}),
+			expectedTip: big.NewInt(30 * params.GWei),
+		},
```

Comment:
It feels a bit weird that we aren't testing against the default config here... but I think that's ok.